Userland/Utilities/pcfconv.cpp

```diff
@@ -0,0 +1,643 @@
+/*
+ * Copyright (c) 2024, Julian Offenhäuser <offenhaeuser@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Math.h>
+#include <AK/StringBuilder.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibGfx/Font/BitmapFont.h>
+#include <LibGfx/Font/FontStyleMapping.h>
+#include <LibMain/Main.h>
+#include <LibURL/URL.h>
+#include <limits.h>
+#include <unistd.h>
+
+#define PCF_GLYPH_PAD_MASK (3 << 0)
+#define PCF_BYTE_MASK (1 << 2)
+#define PCF_BIT_MASK (1 << 3)
+#define PCF_SCAN_UNIT_MASK (3 << 4)
+
+#define PCF_PROPERTIES (1 << 0)
+#define PCF_ACCELERATORS (1 << 1)
+#define PCF_METRICS (1 << 2)
+#define PCF_BITMAPS (1 << 3)
+#define PCF_INK_METRICS (1 << 4)
+#define PCF_BDF_ENCODINGS (1 << 5)
+#define PCF_SWIDTHS (1 << 6)
+#define PCF_GLYPH_NAMES (1 << 7)
+#define PCF_BDF_ACCELERATORS (1 << 8)
+
+#define PCF_DEFAULT_FORMAT 0x00000000
+#define PCF_INKBOUNDS 0x00000200
+#define PCF_ACCEL_W_INKBOUNDS 0x00000100
+#define PCF_COMPRESSED_METRICS 0x00000100
+
+class PCFFile : public RefCounted<PCFFile> {
+public:
+    static ErrorOr<NonnullRefPtr<PCFFile>> create(ReadonlyBytes);
+
+    ErrorOr<String> construct_filename() const;
+
+    Optional<i16> glyph_index_for(u16 code_point) const;
+    ErrorOr<void> draw_glyph(u16 index, Gfx::GlyphBitmap&) const;
+    u8 glyph_width(u16 index) const { return m_glyphs.at(index).width; }
+    u8 baseline() const;
+    size_t highest_codepoint() const;
```

Comment:
Can this be larger than u16? The mapping only has 16bit values for min and max afaict.